pkg/oci/build.go

```diff
@@ -208,6 +210,13 @@ func createImageIndex(ctx context.Context, target oras.Target, o *BuildGadgetIma
 		}
 		layers = append(layers, manifestDesc)
 	}
+	if o.WasmObjectPath != "" {
+		manifestDesc, err := createManifestForTarget(ctx, target, o.MetadataPath, o.WasmObjectPath, "wasm")
```

Comment:
One issue to fix:

`createManifestForTarget` uses the following ebpf mediatype instead of a wasm mediatype:

```go
eBPFObjectMediaType = "application/vnd.gadget.ebpf.program.v1+binary"
```

So the manifest looks like:
```json
{
  "schemaVersion": 2,
  "config": {
    "mediaType": "application/vnd.gadget.config.v1+yaml",
    "digest": "sha256:56077f72620cce86e2c59aa3b0f0fb77764ca9e5d87c2a1cc7ac01459533934c",
    "size": 2057,
    "annotations": {
      "org.opencontainers.image.title": "config.yaml"
    }
  },
  "layers": [
    {
      "mediaType": "application/vnd.gadget.ebpf.program.v1+binary",
      "digest": "sha256:cc75bbd53264752cdad8ea56b004a71efdad52f7014bbf1c1f651877ef3a0661",
      "size": 19944,
      "annotations": {
        "org.opencontainers.image.authors": "TODO: authors",
        "org.opencontainers.image.description": "TODO: description",
        "org.opencontainers.image.title": "program.o"
      }
    }
  ]
}
```

Reply:
I can file a follow-up issue for this.

Reply:
> What about modifying the type depending on the target?

Thanks! I used that solution for simplicity.